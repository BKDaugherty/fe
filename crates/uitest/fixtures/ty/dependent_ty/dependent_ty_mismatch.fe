pub struct Foo<const N: u32> {
    N: u256
}

pub struct Foo2<T, const N: u32> {
    t: T,
    N: u256
}


pub struct Bar<const N: bool> {
    N: bool
}

pub fn foo(x: Foo<true>) {}
pub fn foo2(x: Foo2<i32, false>) {}

pub fn bar(x: Bar<3>) {}