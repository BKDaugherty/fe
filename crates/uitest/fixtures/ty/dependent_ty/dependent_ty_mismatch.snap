---
source: crates/uitest/tests/ty.rs
expression: diags
input_file: crates/uitest/fixtures/ty/dependent_ty/dependent_ty_mismatch.fe
---
error[3-0011]: given type doesn't match the expected dependent type
   ┌─ dependent_ty_mismatch.fe:15:15
   │
15 │ pub fn foo(x: Foo<true>) {}
   │               ^^^^^^^^^ expected `u32` type here, but `bool` is given

error[3-0011]: given type doesn't match the expected dependent type
   ┌─ dependent_ty_mismatch.fe:16:16
   │
16 │ pub fn foo2(x: Foo2<i32, false>) {}
   │                ^^^^^^^^^^^^^^^^ expected `u32` type here, but `bool` is given

error[3-0011]: given type doesn't match the expected dependent type
   ┌─ dependent_ty_mismatch.fe:18:15
   │
18 │ pub fn bar(x: Bar<3>) {}
   │               ^^^^^^ expected `bool` type here, but `<integer>` is given


